debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java

```diff
@@ -391,6 +398,10 @@ else if (data instanceof Double) {
             return ((Double) data).floatValue();
         }
         else if (data instanceof String) {
+            final String s = (String) data;
+            if (isHexToRawFunctionCall(s)) {
```

Comment:
Is it necessary to have condition here and in other instances? Can't the methods be always called and the check and branching done inside of it?

Reply:
One thing I'd like to rework eventually is a different value converter per adapter.  We had talked about this with OpenLogReplicator and it would certainly allow a really good way to overlay specific conversion steps based on another or even configuration within an adapter in a more OO-way.

Reply:
Looking at this more closely, I think we could consolidate many of these, but I don't think it will be universal without potentially looking at using some type of lambda which would be an extra object allocation on the stack for each column converted.  For example, in `convertIntervalYearMonth`, where if it's a hex-to-raw function, we use the return value as-is, but instead, we do some other manipulation since the value is using an alternative format.  The ones where we essentially assign the return value to `data` and pass it through the remainder of the converter method could be changed though.  Is that sufficient you think?

Reply:
I added some changes in 5de55df, let me know if that suffices or if I am off base.